test/chain.js

```diff
@@ -1,15 +1,107 @@
 'use strict';
 
-api.metatests.test('map', (test) => {
+api.metatests.test('for.map', test => {
   const data = [1, 2, 3, 4];
   const expected = [2, 4, 6, 8];
   const fn = (item, callback) => process.nextTick(() => {
     callback(null, item * 2);
   });
+  api.metasync
+    .for(data)
+    .map(fn)
+    .fetch((error, result) => {
+      test.strictSame(result, expected);
+      test.end();
+    });
+});
 
-  api.metasync.for(data).map(fn).fetch((err, result) => {
-    test.error(err, 'must not return an error');
-    test.strictSame(result, expected, `result should be: ${expected}`);
-    test.end();
-  });
+api.metatests.test('for chain sync', test => {
+  api.metasync
+    .for([1, 2, 3, 4])
+    .filter((item, cb) => cb(null, item % 2 === 0))
+    .map((item, cb) => cb(null, item * 2))
+    .reduce((a, b, cb) => cb(null, a + b))
+    .fetch((error, result) => {
+      test.error(error, 'must not return an error');
+      test.strictSame(result, 12); // 2 * 2 + 4 * 2
+      test.end();
+    });
+});
+
+api.metatests.test('for chain async', test => {
+  api.metasync
+    .for([1, 2, 3, 4])
+    .filter((item, cb) => process.nextTick(cb, null, item % 2 === 0))
+    .map((item, cb) => process.nextTick(cb, null, item * 2))
+    .reduce((a, b, cb) => process.nextTick(cb, null, a + b))
+    .fetch((error, result) => {
+      test.error(error, 'must not return an error');
+      test.strictSame(result, 12); // 2 * 2 + 4 * 2
+      test.end();
+    });
+});
+
+api.metatests.test('for chain error', test => {
+  api.metasync
+    .for([1, 2, 3, 4])
+    .filter((item, cb) => cb(null, item % 2 === 0))
+    .map((item, cb) => cb(new Error('Something happens')))
+    .reduce((a, b, cb) => cb(null, a + b))
+    .fetch((error, result) => {
+      test.strictSame(error instanceof Error, true);
```

Comment:
Wdyt of adding `test.isError()`  or smth similar to metatests for such cases?

Reply:
yes, it would be good

Reply:
Well, easy enough, done: https://github.com/metarhia/metatests/pull/66/commits/7fa5e9c7829bc4e9c16c21838aa7c6fbb39e3e13 =)